### PR TITLE
Add support for hint_locals

### PR DESCRIPTION
**Pull request by fmoletta**
Add support for hint_locals
hint_locals is a dictionary of python objects that can be accessed and modified by any hint regardless of scopes

### CODE REVIEW

**Review by entropidelic — COMMENTED** [review]
Good work 😎. Just a little detail regarding a cast and it should be approved

File: src/ids.rs [hunk 1]
```diff
@@ -123,7 +123,7 @@ pub fn compute_addr_from_reference(
         None => return Err(to_py_error(VirtualMachineError::NoRegisterInReference)),
     };
     if hint_reference.offset1.is_negative()
-        && base_addr.offset < hint_reference.offset1.abs() as usize
+        && base_addr.offset < hint_reference.offset1.unsigned_abs() as usize
```
[the remark this answers is not in the document]
Comment (fmoletta) [reply src/ids.rs]: Solved!

**Review by entropidelic — APPROVED** [review]

**Comment by entropidelic** [comment]
LGTM!

File: src/cairo_run.rs [hunk 2]
```diff
@@ -23,6 +26,7 @@ pub fn cairo_run_py<'a>(
     print_output: bool,
     trace_file: Option<&str>,
     memory_file: Option<&str>,
+    hint_locals: Option<HashMap<String, PyObject>>,
```
Comment (Oppen) [inline src/cairo_run.rs:29]: In which cases would this be `None`? It looks like we could simply assume one to be present, even if an empty one, to have code be a little bit simpler.

**Review by Oppen — APPROVED** [review]

File: src/cairo_run.rs (hunk 2 above)
Comment (fmoletta) [reply src/cairo_run.rs:29]: Option is used so that hint_locals doesn't need to be specified when calling the function from python

Comment (Oppen) [reply src/cairo_run.rs:29]: Nice. I didn't know PyO3 handled that too.

**Pull request merged**